Introduce new `container` integrations (#10577)
DataDog/integrations-core: container/README.md

> # Agent Check: Containerd

Reviewer: ```suggestion
# Agent Check: Container
```
this might not be correct, but you refer to it as "Container" everywhere else

## Overview

This check reports a set of metrics about any running containers, regardless of the runtime used to start them.

## Setup

### Installation

Container is a core Agent 6 check and is automatically activated if any supported container runtime is detected.
Configuring access to supported container runtimes (Docker, containerd) may be required depending on your environment.

#### Installation on containers

> The `container` check requires some folders to be mounted to allow for automatic activation. This is handled by our official Helm Chart, the Datadog Operator as well as documented setups for Kubernetes, Docker, ECS and ECS Fargate.

Reviewer: ```suggestion
The `container` check requires some folders to be mounted to allow for automatic activation. This is handled by our official Helm Chart, the Datadog Operator, and as documented set ups for Kubernetes, Docker, ECS, and ECS Fargate.
```

### Configuration

> Currently, the `container` check does not expose any specific configuration setting. However to customize common fields or to force the activation of the `container` check, follow these steps:

Reviewer: ```suggestion
Currently, the `container` check does not expose any specific configuration settings. To customize common fields or to force the activation of the `container` check, follow these steps:
```

> 1. Create the `container.d/conf.yaml` file in the `conf.d/` folder at the root of your Agent's configuration directory.

Reviewer 2: I don't see a `spec.yaml` or `conf.yaml.sample` file for this integration. Does it set any configs?

Author: The `conf.yaml` is the `datadog-agent` repo as it's core (Go) check.

> 1. Create the `container.d/conf.yaml` file, in the `conf.d/` folder at the root of your Agent's configuration directory.

Reviewer: ```suggestion
1. Create the `container.d/conf.yaml` file in the `conf.d/` folder at the root of your Agent's configuration directory.
```

> 2. [Restart the Agent][2]

Reviewer 2: Small nit, but it looks like `[2]` doesn't actually link to instructions on restarting the agent.

### Validation

> [Run the Agent's `status` subcommand][3] and look for `container` under the Checks section.

Reviewer: ```suggestion
[Run the Agent's `status` subcommand][3] and look for `container` under the **Checks** section.
```

## Data Collected

### Metrics

> The `container` check can collect metrics about CPU, Memory, Network and Disks IO.

Reviewer: this section won't render in the docs. The metrics section has to only contain the `See [metadata.csv][4] for a list of metrics provided by this integration.` sentence and it generates the table, anything else gets overwritten. Add this as a note in the validation or configuration sections.

#### Installation on containers

The `container` check requires some folders to be mounted to allow for automatic activation. This is handled by our official Helm Chart, the Datadog Operator, and as documented set ups for Kubernetes, Docker, ECS, and ECS Fargate.

### Configuration

Currently, the `container` check does not expose any specific configuration settings. To customize common fields or to force the activation of the `container` check, follow these steps:

1. Create the `container.d/conf.yaml` file in the `conf.d/` folder at the root of your Agent's configuration directory.

2. [Restart the Agent][3]

The `container` check can collect metrics about CPU, Memory, Network and Disks IO.
Some metrics may not be available depending on your environment (Linux / Windows, for instance).

### Validation

[Run the Agent's `status` subcommand][3] and look for `container` under the **Checks** section.

## Data Collected

### Metrics

See [metadata.csv][4] for a list of metrics provided by this integration.

## Troubleshooting

Need help? Contact [Datadog support][2].

> [2]: https://docs.datadoghq.com/help/

Reviewer 2: Small nit, the links should start at `[1]`. You can fix this with `ddev validate readmes container -fl`
